File: frontend/src/types/DataMaster/Kelas.ts
export type KelasFormValues = {
  tingkat_kelas: number | "";
  nama_kelas: string;
};

export type TingkatKelas = {
  id_tingkat_kelas: number;
  tingkat_kelas: number;
};
export type NamaKelas = {
  id_nama_kelas: number;
  id_tingkat_kelas: number;
  nama_kelas: string;
};

export type FullDataKelas = {
  item_tingkat_kelas: TingkatKelas[];
  item_nama_kelas: NamaKelas[];
};

export type KelasFilterParams = {
  search?: string;
  tingkatKelas?: number;
};

export type KelasSubmitResponse = {
  id: number;
};

File: backend/internal/core/domain/kelas/kelas.go
package kelas

type ID int

type TingkatKelas struct {
	IdTingkatKelas ID
	TingkatKelas   int
}

type NamaKelas struct {
	IdNamaKelas    ID
	IdTingkatKelas ID
	NamaKelas      string
}

type FullKelasData struct {
	ItemsTingkatKelas []TingkatKelas
	ItemsNamaKelas    []NamaKelas
}

type KelasData struct {
	ItemsTingkatKelas TingkatKelas
	ItemsNamaKelas    NamaKelas
}
